core/domain/html_cleaner.py

```diff
@@ -191,3 +192,371 @@ def get_rte_components(html_string: str) -> List[ComponentsDict]:
+        ValidationError. Video autoplay-with-value attribute should be boolean.
+        ValidationError. Video does not have video_id-with-value attribute.
+        ValidationError. Link does not have text-with-value attribute.
+        ValidationError. Link does not have url-with-value attribute.
+        ValidationError. Link url-with-value attribute should not be empty.
+        ValidationError. Math does not have math_content-with-value attribute.
+        ValidationError. Math math_content-with-value attribute should not be
+            empty.
+        ValidationError. Math does not have raw_latex-with-value attribute.
+        ValidationError. Math raw_latex-with-value attribute should not be
+            empty.
+        ValidationError. Math does not have svg_filename-with-value attribute.
+        ValidationError. Math svg_filename-with-value attribute should not be
+            empty.
+        ValidationError. Math svg_filename attribute does not have svg
+            extension.
+        ValidationError. Tabs tag present inside another tabs or collapsible.
+        ValidationError. Collapsible tag present inside tabs or another
+            collapsible.
+    """
+    soup = bs4.BeautifulSoup(html_data, 'html.parser')
+    empty_values = ['&quot;&quot;', '', '\'\'', '\"\"']
+    for tag in soup.find_all('oppia-noninteractive-image'):
+        if not tag.has_attr('alt-with-value'):
+            raise utils.ValidationError(
+                'Image tag does not have \'alt-with-value\' attribute.'
+            )
+
+        alt_value = utils.unescape_html(
+            tag['alt-with-value'])[1:-1].replace('\\"', '')
+        if len(alt_value.strip()) < 5:
+            raise utils.ValidationError(
+                'The length of the image tag \'alt-with-value\' '
+                'attribute value should be at least 5 characters.'
+            )
+
+        if not tag.has_attr('caption-with-value'):
+            raise utils.ValidationError(
+                'Image tag does not have \'caption-with-value\' attribute.'
+            )
+
+        caption_value = utils.unescape_html(
+                tag['caption-with-value'])[1:-1].replace('\\"', '')
+        if len(caption_value.strip()) > 500:
+            raise utils.ValidationError(
+                'Image tag \'caption-with-value\' attribute should not '
+                'be greater than 500 characters.'
+            )
+
+        if not tag.has_attr('filepath-with-value'):
+            raise utils.ValidationError(
+                'Image tag does not have \'filepath-with-value\' attribute.'
+            )
+
+        filepath_value = utils.unescape_html(
+                tag['filepath-with-value'])[1:-1].replace('\\"', '')
+        if filepath_value.strip() in empty_values:
+            raise utils.ValidationError(
+                'Image tag \'filepath-with-value\' attribute should not '
+                'be empty.'
+            )
```

Comment:
Can this be parametrized and put into a function.

Reply:
Done